### DISCUSSION TITLE
"No visible binding for global variable" in tandem with "Undefined global functions or variables"

### DISCUSSION THREAD

**Discussion by maxl-amberg** [ThinkR-open/golem#660] [Q&A] [answered]
Hi,
I've got another question regarding one of my apps which runs perfectly fine when I use "run_dev.R" stored in the "dev" folder, but which always renders the following NOTE when I run devtools::check() in "03_deploy.R": 

_*checking R code for possible problems ... NOTE_

_mod_dataviz_server: no visible binding for global variable 'country'_

_Undefined global functions or variables: country_

This warning does not only pop up for the variable country but for all other variables stored in the two datasets df_final_long and df_final_wide, which are both included in globals.R by writing the following:
_globalVariables(
  c("df_final_wide",
    "df_final_long")
)_

@ColinFay : Is it because one can only include one dataset here?
Or would e.g. using _country <- rlang::sym(input$country)_ along with _!!country_ instead of using _df_final_long$country_ solve this issue?

Again, thanks a lot in advance!

**Comment by VincentGuyader** [comment, marked as answer]
Hi, 

this package could help you to create your `globalVariables` call.
https://github.com/ThinkR-open/checkhelper

Regards

**Reply by maxl-amberg** [reply]
Hi,

perfect! Copying the code (suggested after running `checkhelper::print_globals(quiet = TRUE)`) to my globals.R file, removed the note.
Thanks a lot!

Best

**Comment by ColinFay** [comment]
Hey, 

As @VincentGuyader pointed out, this is an issue that can be solved via `{checkhelper}`.

These error are usually due to using non standard evaluation in tidyverse functions. 

You can either define the variables as global variables, or use the `.data` pronoun in tidyverse functions, as described in https://dplyr.tidyverse.org/articles/programming.html 

Cheers, 
Colin

**Reply by maxl-amberg** [reply]
Hey,

Thanks for giving additional background information.

Best,
Maxl
